@AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fn_getInput.sqf

```diff
@@ -92,22 +94,39 @@ if (_flightModel == "SFMPlus") then {
         _tailRtrFixed = true;
     };
     
+    private _paused   = isNull findDisplay 49;
+    private _chatting = isNull findDisplay 24;
+    private _inDialog = !dialog;
+    private _isZeus   = !isNull findDisplay 312;
+
     if (!_hydFailure || _emerHydOn) then {
         if (isNil "fza_sfmplus_collectiveOutput") then {
             fza_sfmplus_collectiveOutput = 0;
         };
 
         if (fza_ah64_sfmPlusKeyboardOnly) then {
-            private _collectiveVal = fza_sfmplus_collectiveOutput;
             if (_keyCollectiveUp > 0.1) then { _collectiveVal = _collectiveVal + ((1.0 / 3.0) * _deltaTime); };
             if (_keyCollectiveDn > 0.1) then { _collectiveVal = _collectiveVal - ((1.0 / 3.0) * _deltaTime); };
-            fza_sfmplus_collectiveOutput = [_collectiveVal, 0.0, 1.0] call bis_fnc_clamp;
+            _collectiveVal = [_collectiveVal, 0.0, 1.0] call bis_fnc_clamp;
+
+            private _isPlaying  = _paused && _chatting && _inDialog && !_isZeus;
+
+            if (isNil "fza_sfmplus_prevCollective" || isNil "fza_sfmplus_lastIsPlaying") then {
+                fza_sfmplus_collectiveOutput = _collectiveVal;
+            } else {
+                if (_isPlaying && fza_sfmplus_lastIsPlaying) then {
+                    fza_sfmplus_collectiveOutput = fza_sfmplus_prevCollective;
+                };
+            };
+
+            fza_sfmplus_lastIsPlaying  = _isPlaying;
+            fza_sfmplus_prevCollective = _collectiveVal;
         } else {
-            private _collectiveVal = _joyCollectiveUp - _joyCollectiveDn;
+            _collectiveVal = _joyCollectiveUp - _joyCollectiveDn;
             _collectiveVal = [_collectiveVal, -1.0, 1.0] call BIS_fnc_clamp;
             _collectiveVal = linearConversion[ -1.0, 1.0, _collectiveVal, 0.0, 1.0];
 
-            private _isPlaying  = isNull findDisplay 49 && isNull findDisplay 24 && !dialog;
+            private _isPlaying  = _paused && _chatting && _inDialog && !_isZeus;
```

Comment:
If this is needed in both sides of the if statement - can't we put it outside of it?

Reply:
Tried it that way, didn’t work, not sure why